packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerHighlight/utils/textBoxMapping/TextProvider.ts

```diff
@@ -0,0 +1,107 @@
+import { TextSpan } from '../../types';
+import {
+  END,
+  START,
+  spanIntersects,
+  spanIncludesIndex,
+  spanGetText,
+  spanIntersection
+} from '../../../../utils/textSpan';
+import { findLargestIndex } from '../common/findLargestIndex';
+
+const MAX_HISTORY = 3;
+
+export type TextMatch = {
+  /** matched text span */
```

Comment:
For single line comments like these, let's keep with convention of just starting with `//` and leave `/**` for multi-line comments (here and elsewhere in PR).

(Wonder if there's an eslint/prettier plugin for this?)